lib/LaTeXML/Package/authblk.sty.ltxml

```diff
@@ -51,19 +60,25 @@ DefConstructor('\affil[]{}',
     $whatsit->setProperty(mark => ToString($mark));
     return; });
 
-Tag('ltx:note',     afterClose => \&authblkRelocateAffil);
-Tag('ltx:document', afterClose => \&pruneEmptyAffil);
+Tag('ltx:document', afterClose => \&authblkRelocateAffil);
 
 sub authblkRelocateAffil {
-  my ($document, $affilnode) = @_;
-  if (($affilnode->getAttribute('role') || '') eq 'affiliationtext') {
-    if (my $mark = $affilnode->getAttribute('mark')) {
+  my ($document)   = @_;
+  my @affil_nodes  = $document->findnodes(".//ltx:note[\@role='affiliationtext']");
+  my @author_nodes = $document->findnodes(".//ltx:contact[\@role='affiliationmark' and \@_mark]");
+  for my $affil_node (@affil_nodes) {
+    # first, unbind all nodes - they get reparented or discarded.
+    $affil_node->parentNode->removeChild($affil_node);
+    # next, see if we can find a new home.
+    if (my $affil_mark = $affil_node->getAttribute('mark')) {
       # Find authors with same mark.
-      my @authors = $document->findnodes(".//ltx:contact[\@role='affiliationmark' and \@_mark='$mark']");
-      foreach my $author (@authors) {
-        $document->appendClone($author, $affilnode->childNodes);
-        $author->setAttribute(role => 'affiliation'); }
-      $affilnode->parentNode->removeChild($affilnode); } }
+      for my $author_node (@author_nodes) {
+        my @author_marks = split(/\s*,\s*/, ($author_node->getAttribute('_mark') || ''));
+        next unless (grep { $_ eq $affil_mark } @author_marks);
+        # An affiliation node may be reused across authors,
+        # but should still get unlinked the first time we reparent it
+        $document->appendClone($author_node, $affil_node);
+        $author_node->setAttribute(role => 'affiliation'); } } }
```

Comment:
So, the affiliations are created as `ltx:note` because they haven't (yet) got an author to contain them (in which case they would just be `ltx:contact`); right? But when you're copying the affiliation into the author contact, it should be the contents of the note, not the whole note.